packages/imgareaselect/src/index.js

```diff
@@ -0,0 +1,1236 @@
+       * available (so that it can be obtained in the callback).
+       */
+      setTimeout(function () { options.onInit(img, getSelection()); }, 0);
+    }
+
+    /**
+     * Document keypress event handler
+     *
+     * @param event
+     *            The event object
+     * @return false
+     */
+    var docKeyPress = function(event) {
+      var k = options.keys, d, t, key = event.keyCode;
+
+      d = !isNaN(k.alt) && (event.altKey || event.originalEvent.altKey) ? k.alt :
+        !isNaN(k.ctrl) && event.ctrlKey ? k.ctrl :
+          !isNaN(k.shift) && event.shiftKey ? k.shift :
+            !isNaN(k.arrows) ? k.arrows : 10;
+
+      if (k.arrows == 'resize' || (k.shift == 'resize' && event.shiftKey) ||
+        (k.ctrl == 'resize' && event.ctrlKey) ||
+        (k.alt == 'resize' && (event.altKey || event.originalEvent.altKey)))
+      {
+        /* Resize selection */
+
+        switch (key) {
+          case 37:
+            /* Left */
+            d = -d;
+          case 39:
+            /* Right */
+            t = max(x1, x2);
+            x1 = min(x1, x2);
+            x2 = max(t + d, x1);
+            fixAspectRatio();
+            break;
+          case 38:
+            /* Up */
+            d = -d;
+          case 40:
+            /* Down */
+            t = max(y1, y2);
+            y1 = min(y1, y2);
+            y2 = max(t + d, y1);
+            fixAspectRatio(true);
+            break;
+          default:
+            return;
+        }
+
+        doResize();
+      }
+      else {
+        /* Move selection */
+
+        x1 = min(x1, x2);
+        y1 = min(y1, y2);
+
+        switch (key) {
+          case 37:
+            /* Left */
+            doMove(max(x1 - d, left), y1);
+            break;
+          case 38:
+            /* Up */
+            doMove(x1, max(y1 - d, top));
+            break;
+          case 39:
+            /* Right */
+            doMove(x1 + min(d, imgWidth - selX(x2)), y1);
+            break;
+          case 40:
+            /* Down */
+            doMove(x1, y1 + min(d, imgHeight - selY(y2)));
+            break;
+          default:
+            return;
+        }
+      }
+
+      return false;
+    };
+
+    /**
+     * Apply style options to plugin element (or multiple elements)
+     *
+     * @param $elem
+     *            A jQuery object representing the element(s) to style
+     * @param props
+     *            An object that maps option names to corresponding CSS
+     *            properties
+     */
+    function styleOptions($elem, props) {
+      for (var option in props)
+        if (options[option] !== undefined)
+          $elem.css(props[option], options[option]);
+    }
+
+    /**
+     * Set plugin options
+     *
+     * @param newOptions
+     *            The new options object
+     */
+    function setOptions(newOptions) {
+      if (newOptions.parent)
+        ($parent = $(newOptions.parent)).append($box.add($outer));
```

Comment:
GitHub reports a potential XSS vulnerability here, here's how we can fix it.

Not related to this original bug of course, but I also don't want to simply dismiss this issue here.

So let's just fix it.

```suggestion
        ($parent = $.find(newOptions.parent)).append($box.add($outer));
```